### DISCUSSION TITLE
What do you think about my Filelist extension?

### DISCUSSION THREAD

**Discussion by GiovanniSalmeri** [datenstrom/community#418] [Ask a question] [answered]
Sorry for flooding this space of issues with my little contributions... I have published yet another small extension: [Filelist](https://github.com/GiovanniSalmeri/yellow-filelist). Its purpose is simply to show (recursively) a list of files (typically documents, forms etc.) for download. Creating a shortcut [filelist folder] and setting an FTP folder is moreover probably the easiest way to allow someone (also with zero experience) to publish files. I wrote this piece of code several years ago, now I have modernised and adapted it to the great Yellow.

**Comment by schulle4u** [comment, marked as answer]
I'm glad no longer to be the only one posting extensions every few weeks. 😀

The filelist sounds interesting, I'll give it a try. What is the advantage of allowing forbidden characters in filenames? They possibly will look a little prettier on the website, but from a user's perspective your solution for this sounds a little to technical I think. Maybe an optional metadata YAML file for each file in a folder could be more user friendly.

**Comment by GiovanniSalmeri** [comment]
@schulle4u Thank you very much for your comment. As regards the forbidden characters: I agree that the solution is not ideal, but it is also true that those characters are very seldom used or needed in link text, so any solution will have a very little impact on the general usability. Anyway I find your idea of an optional metadata file very good, I will add that as soon as possible (perhaps with a setting for choosing between the two methods). Thank you!

**Comment by GiovanniSalmeri** [comment]
@schulle4u I have added just now the system you proposed, with a small simplification: not a YAML file, but a simple text file with the extension `.text` and the text to be shown as only content. The previous encoding system does not conflict, but anyway can be disabled. Thank you again!

**Comment by schulle4u** [comment]
Hi @GiovanniSalmeri,

I recently re-discovered your extension and am now using it in [one of my projects](https://m45.dev/en/downloads/). 👍 As I wanted to display more information in the file list, I modified it slightly. There is now a MetaData setting where you can specify additional information to be displayed in the file list using template variables. Currently, these are the file type, file size, and file modification date. I have sent this to you as a [pull request](https://github.com/GiovanniSalmeri/yellow-filelist/pull/4) in case you find it useful and would like to incorporate it. 🙂

Steffen

**Reply by GiovanniSalmeri** [reply]
Thank you very much and sorry for the late answer. Merged! 👍
